src/Blocks/BlockLeaves.h

```diff
@@ -40,22 +40,33 @@ class cBlockLeavesHandler :
 	{
 		cFastRandom rand;
 
-		// Old leaves - 3 bits contain display; new leaves - 1st bit, shifted left two for saplings to understand
-		if (rand.NextInt(20) == 0)
+		// There is a chance to drop a sapling that varies depending on the type of leaf broken.
+		// TODO: Take into account fortune for sapling drops.
+		int chance;
+		if ((m_BlockType == E_BLOCK_LEAVES) && ((a_BlockMeta & 0x03) == E_META_LEAVES_JUNGLE))
+		{
+			// Jungle leaves have a 2.5% chance of dropping a sapling.
+			chance = rand.NextInt(40);
+		}
+		else
+		{
+			// Other leaves have a 5% chance of dropping a sapling.
+			chance = rand.NextInt(20);
+		}
+		if (chance == 0)
 		{
 			a_Pickups.push_back(
-				cItem(
-					E_BLOCK_SAPLING,
-					1,
-					(m_BlockType == E_BLOCK_LEAVES) ? (a_BlockMeta & 0x03) : (2 << (a_BlockMeta & 0x01))
-				)
-			);
+			cItem(
+				E_BLOCK_SAPLING,
+				1,
+				(m_BlockType == E_BLOCK_LEAVES) ? (a_BlockMeta & 0x03) : (4 + (a_BlockMeta & 0x01))
+			));
```

Comment:
The indent is wrong here. Either merge the `a_Pickups.push_back` and `cItem(` lines, or separate the two parentheses here (and indent the internals by one more level)